be/src/exec/spill/formatter.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/spill/formatter.h"
+
+#include "exec/spill/spiller.h"
+#include "gutil/port.h"
+#include "runtime/runtime_state.h"
+#include "serde/column_array_serde.h"
+
+namespace starrocks {
+namespace spill {
+
+class ColumnarFormatter : public Formatter {
+public:
+    ColumnarFormatter(ChunkBuilder chunk_builder, const BlockCompressionCodec* compress_codec)
+            : _chunk_builder(std::move(chunk_builder)), _compress_codec(compress_codec) {}
+    ~ColumnarFormatter() = default;
```

Comment:
1. compression is time-consuming, prefer encoding like the exchange.
2. we may to unify the compression and encoding in here and exchange.

Reply:
> 1. compression is time-consuming, prefer encoding like the exchange.
> 2. we may to unify the compression and encoding in here and exchange.

make sense. we can use encoding here, I had this idea before, but now the interface of encoding is not universal enough. it relys on PB somewhere. I will unify them in the following PR